### DISCUSSION TITLE
Need help with "Missing required ionCube Loader" and PHP versions

### DISCUSSION THREAD

**Discussion by mtbluedog** [geodesicsolutions-community/geocore-community#223] [Q&A] [answered]
Hi,

I have two strange issues which I hope someone can help with, mostly PHP versions, is appears.

I inherited this site - GeoCore MAX Ver. 18.02.0 - and am trying to get it working again; it worked fine until I think the webhost, GoDaddy, force upgraded the Cpanel account to PHP5.6, where it is now. I can't downgrade to an earlier PHP version. 

The first error is that the home page shows "Missing required ionCube Loader." In Cpanel, PHP5.6 is selected, with the PHP extension ioncube_loader (which has been on by default). I've tried toggling `ioncube_loader_4`  and` ioncube_loader_5` with no luck.

Ioncube is included in public_html:
 
![ioncube](https://github.com/geodesicsolutions-community/geocore-community/assets/23107911/93e53f15-ae87-4970-8ef2-0746571f256e)

phpinfo in public_html shows Ioncube:
Loader version 	12.0.5
ionCube24 features 	unconfigured 
ioncube.loader.encoded_paths	no value

Any ideas why Ioncube complains about not being loaded?

The second error is at example.com/admin/ . I get the login screen, but when I do log in, I get the error 

 `--> Smarty: Invalid compiled template for 'validating_login.tpl' <-- `

The file validating_login.tpl exists.

I tried changing to PHP7.4 in Cpanel, but that doesn't help; "Missing required ionCube Loader" still shows, and /admin/ throws these different errors, which appear to mean that the mysql_connect function is deprecated under PHP5.6

```
Error: Call to undefined function mysql_connect() in /home/ofzilgywzxa4/public_html/classes/adodb/drivers/adodb-mysql.inc.php:461
```
Are these PHP version errors? Something else?

**I'll appreciate any help. Thanks!**

**Comment by vicos59** [comment]
If it is a cpanel server, doesn't the PHP instance have to be built/configured to use ionCube in WHM ?

**Reply by mtbluedog** [reply]
Thanks, it seems so, but there is no WHM access in the hosting account. It's Economy Hosting and there's nothing more than GoDaddy's limited version of Cpanel. I can turn on the PHP extensions ioncube_loader (which has been on by default) and ioncube_loader_4 and ioncube_loader_5, but that doesn't help.

**Reply by mtbluedog** [reply]
Well, that's interesting; PHP5.6 and the site work now. The backup restore I had to do took longer for some reason.

I still have the second error is at example.com/admin/. I get the login screen, but when I log in, I get a white screen and the error
`--> Smarty: Invalid compiled template for 'validating_login.tpl' <-- `

Does GeoCore use Smatry, the template engine?

This GeoCore site was installed by the original commercial company in 2018.

**Reply by vicos59** [reply]
Yes, the entire system is based on Smarty.

**Reply by vicos59** [reply]
Overall, your best bet is try and install the Patch to remove the licensing and then upgrade to the most recent version. I would use PHP 7.4 if you get to that point. 8.1 should work, but there are likely some things that will need fixed that we missed.

**Reply by mtbluedog** [reply]
Thanks, I wanted to try the patch, but the remove the licensing patch link on https://geodesicsolutions.org/resources.php now 404's.
Is there anywhere else I can get the patch?

**Reply by vicos59** [reply]
I uploaded to the other thread you created here on github last night. I think it was you, anyway...

**Comment by vicos59** [comment]
[unencoded_unlicense_patch_18.02.0.bff354dbba748dd30c1d9a7d1ccc7dd61a1b95fc.zip](https://github.com/geodesicsolutions-community/geocore-community/files/13891155/unencoded_unlicense_patch_18.02.0.bff354dbba748dd30c1d9a7d1ccc7dd61a1b95fc.zip)

**Comment by mtbluedog** [comment]
Thanks! I didn't see your other post.

**Comment by blufyremedia** [comment, marked as answer]
@mtbluedog for your two issues:

Firstly, you've posted sensitive login information to your database from your error dump.  I've removed the relevant parts, but you should cycle your database credentials regardless.

**### IonCube**
Once you apply the license patch this issue should go away. The patch replaces the two encoded files that originally required Ioncube loaders, and unless you have third-party add-ons or code that also require Ioncube, you should no longer need the loaders.

That aside, Ioncube loaders are something that are configured for each PHP version. Your host, or you, should be able to configure these at the hosting level. Regardless, if you do still need the loaders - it would need to set up for the desired PHP version. It sounds like Ioncube is configured for PHP 5.6, but not your other versions.  You mentioned that you use GoDaddy, so see this article:

https://ca.godaddy.com/help/install-ioncube-in-my-linux-hosting-account-26907


**### validating_login.tpl**
We've come across this error on other sites using GoDaddy.  Simply replace the affected file with the original from the original software package.

`admin/templates/validating_login.tpl`

**Comment by mtbluedog** [comment]
oops :( Thanks, I changed the database user and password.
And yes, nothing else in the hosting account uses Ioncube, so I'll try the patch.
